### ISSUE TITLE
Can we apply these codes to the regression problem?

### ISSUE THREAD

**Issue by hanamthang** [alperbulbul1/Feature-Selection-with-Firefly-Algorithm#1]
Hello, 

Can I use your code for regression problem with continuous variables? It seems only binary variable can be used at the moment. 

Regards,
Thang

**Comment by alperbulbul1** [comment]
Hi, 

You can use any regressor instead of the support vector machine classifier in this code. You will need at least two test samples to calculate the R ^ 2 value. Therefore, instead of leaveoneout cross validation, you can write cv = 5 or any number of splits you want.

**Comment by hanamthang** [comment]
Thank you for the comments. I can run your FFA code for feature selection in regression now. 
Could you please explain a bit more: 

1. Is this the number of iteration? 
`self.nFES = 1  # number of function evaluations (repeate number)`

2. From: 
`def evaluation(feature_possibilities):
        feature_possibilities = np.round(feature_possibilities)

        feature_possibilities = feature_possibilities > np.float32(0.5)

        selectedX = X[:, feature_possibilities]
        
        from sklearn.ensemble import RandomForestRegressor as rfr
        rfr = rfr(n_estimators=100, min_samples_leaf=1, min_samples_split=3)
      
        evaluation = cross_val_score(rfr, selectedX, y,  cv=5, scoring=mse)

        return evaluation.mean()`

I defined a regression model (Random Forest) here using the mean square error during the CV = 5. 

How can I extract the MSE values during the CV? Can I plot the iterations with the MSE? 

How should I understand:
`        selectedX = X[:, feature_possibilities]`
here. selectedX is different from bestX in which bestX is selected features after FFA.

Many thanks,
Thang

**Comment by fatima24-a** [comment]
name 'gene_take_or_not' is not defined. I got this error when I run this code. that's why I can't final accuracy. kindly help me out how it will be fix?
